### ISSUE TITLE
persisted queries with transform doesn't work with fragments

### ISSUE THREAD

**Issue by danilobuerger** [gajus/babel-plugin-graphql-tag#59]
When using transform to calculate the hash for the persisted query, the whole query including fragments has to be known. However, both source and ast parameters for transform only contain the query, not the fragments.

**Comment by shmax** [comment]
@danilobuerger  Hey there, I only just noticed this issue, but I was the one that added the transform stuff back then, and I'm trying to update my own code to use fragments. Before I dive in and start trying to fix this, I was wondering if you had any suggestions?

**Comment by danilobuerger** [comment]
I ended up not using this plugin.

**Comment by shmax** [comment]
Gotcha. Can I ask what solution you went with?

**Comment by danilobuerger** [comment]
https://github.com/dotansimha/graphql-code-generator

**Comment by shmax** [comment]
Right, I'm using that too, but how did you extract the graphql queries and replace them with query ids?

**Comment by shmax** [comment]
Also, did you find any way use the string template style?

```
  var document =  gql`query... `;
```

**Comment by danilobuerger** [comment]
With a custom plugin (not open sourced). Sorry I don't have the time to go into more details.

**Comment by shmax** [comment]
No worries, that's good enough. I have a custom plugin, myself. Thanks.
